scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -581,7 +581,8 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
                   // if does not work with indentation in pattern matches
                   val shouldNotIndentIf =
                     prev.is[KwIf] && sepRegions.headOption.exists(r => r == RegionArrow)
-                  !shouldNotIndentIf && prev.is[CanStartIndent] && !next.is[RightBrace]
+                  !shouldNotIndentIf && prev.is[CanStartIndent] && !next.is[RightBrace] &&
+                  !next.is[LeftBracket]
```

Comment:
I was curious why it works for `LeftBrace` without a similar clause and found that the actual reason is in `ExprIntro.unapply` 
It doesn't have `token.is[LeftBracket]` clause.

https://github.com/scalameta/scalameta/blob/a51e009563e69591d3f057e7b777ed4f6a224bc1/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala#L1012-L1021

Reply:
Good catch! I added `!next.is[CantStartStat]` instead

Reply:
Hmm. Sorry for being nonclear.
I think there should be an indent before `RightBrace`.

The sample you tried to fix isn't parsed correctly because of `ExprIntro`. It should have smth like:
 ```scala
  ... || (token.is[LeftBracket] && dialect.allowsPolymorphicFunctions)
```

Without indentation the following will be parsed incorrectly:
```scala
val a: Int = 
  [A] => (x: List[A]) => x.headOption
  42
```

Reply:
Ach! Right! Changed, should be ok now... :sweat_smile: